### PR TITLE
Implement pruner core

### PR DESCRIPTION
**Pull request by Foxcapades**
1. Adds a `Pruner` object that will be the base of the pruning operation from both the background thread and manual pruning trigger endpoint.
2. Adds queries to the Cache-DB component to support the pruning operation.

### CODE REVIEW

File: components/pruner/src/main/kotlin/vdi/component/pruner/Pruner.kt [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      it.deleteImportMessages(datasetID)
+      it.deleteSyncControl(datasetID)
+      it.deleteDataset(datasetID)
+    }
+  }
+
+  private fun DeletedDataset.deleteFromS3(bucket: S3Bucket) {
+    log.debug("deleting dataset {}/{} from S3", ownerID, datasetID)
+
+    bucket.objects.listSubPaths(S3Paths.datasetDir(ownerID, datasetID))
```
Comment (ryanrdoherty) [inline components/pruner/src/main/kotlin/vdi/component/pruner/Pruner.kt:178]: So, there's no such thing as a "directory" right?  Just objects with names that look like paths?

File: settings.gradle.kts [hunk 2]
```diff
@@ -53,6 +53,7 @@ include(":components:kafka")
 include(":components:ldap")
```
Comment (ryanrdoherty) [inline settings.gradle.kts:53]: Above, I added an include for the github checks and packages repo.  You can save some space here by adding:
```
// adds repos for gradle plugin resolution and ensures github creds are provided to the build
apply {
  from("https://raw.githubusercontent.com/VEuPathDB/lib-gradle-container-utils/v4.8.3/includes/common.settings.gradle.kts")
}
```

**Review by ryanrdoherty — APPROVED** [review]

File: components/pruner/src/main/kotlin/vdi/component/pruner/Pruner.kt [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+  /**
+   * Locks the pruner and prunes old/dead datasets.
+   *
+   * This method locks until completion, blocking multiple instances of the
+   * operation being performed simultaneously.  To test if a prune operation is
+   * currently in progress call [isLocked].
+   */
+  fun pruneDatasets() {
+    lock.withLock { doPruning() }
```
[the remark this answers is not in the document]
Comment (Foxcapades) [reply components/pruner/src/main/kotlin/vdi/component/pruner/Pruner.kt:55]: `withLock` is a Kotlin stdlib extra for locking.

File: components/pruner/src/main/kotlin/vdi/component/pruner/Pruner.kt (hunk 1 above)
Comment (Foxcapades) [reply components/pruner/src/main/kotlin/vdi/component/pruner/Pruner.kt:178]: Yep, but some of the S3 allows you to pretend that your keys are paths and we make use of that for "directory" listing.

**Pull request merged**